STM/Sensor/Tanishq/stepper_motor.ino

```diff
@@ -0,0 +1,98 @@
+/*
+This is code to control a stepper mototr. There are two type of controls we have. And that is the direction and the speed/steps per second of the motor.
+We take the input for the number of steps user want per second, the number of seconds it want motor to rotate and finally the direction weather to be clockwise
+or anticlockwise. 
+
+Since motor takes one step corresponding to change in state . We need to calculate these delays using the inputs we got...............
+
+*/
+
+#define STEP_PIN 3                                                                   //Define the pin to output pulse                                                                   
+#define DIR_PIN 4                                                                    //Define the direction pin
+
+
+#define MOTOR_STEPS 200                                                          //Assuming motor steps to be 200 or 1.8degree per step
+
+
+void setup() {
+  
+  pinMode(STEP_PIN, OUTPUT);                                                   //set the pulse pin as output
+  pinMode(DIR_PIN, OUTPUT);                                                    //set the direction pin as output
+
+  Serial.begin(9600);                                                         //Initialise serial monitor, We need it to take user inputs
+}
+
+int rot_speed;
+int duration;                                                                         //define some variables
+int direction_motor;
+
+void loop() {
+
+
+  //  Taking input for the speed of rotation..........................................
+  
+  Serial.print("Enter the steps you want per second (0-");
+  Serial.print(MOTOR_STEPS);
+  rot_speed = Serial.parseInt();
+  if(rot_speed>=0 && rot_speed<=MOTOR_STEPS){                                               //check if speed is not any unusual value. if so tell it to use

```

Comment:
Can you refactor this if block? Is there a better way to write this piece of code?

Reply:
I didn't understand what I need to do.
Is it while(Serial.avaliable==0){}